From e7ce8adc506068491c8545a1990992267598bd33 Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Fri, 4 Apr 2014 17:43:53 +0200
Subject: [PATCH 09/14] kdm: read DesktopNames from session file and export
 XDG_CURRENT_DESKTOP from it

[after converting ';' (XDG list separator) to ':' (path separator)]

As discussed at the freedesktop summit.

XDG_CURRENT_DESKTOP is useful for OnlyShowIn and for the new mimeapps.list spec.
---
 kdm/backend/client.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/kdm/backend/client.c b/kdm/backend/client.c
index 26bb0b4..335cc3e 100644
--- a/kdm/backend/client.c
+++ b/kdm/backend/client.c
@@ -1809,6 +1809,13 @@ startClient(volatile int *pid)
                     if (!strCmp(iniEntry(str, "Desktop Entry", "Hidden", 0), "true") ||
                             !(sessargs = iniEntry(str, "Desktop Entry", "Exec", 0)))
                         sessargs = "";
+                    buf = iniEntry(str, "Desktop Entry", "DesktopNames", 0);
+                    for (buf2 = buf; *buf2; ++buf2) {
+                        if (*buf2 == ';')
+                            *buf2 = ':';
+                    }
+                    userEnviron = setEnv(userEnviron, "XDG_CURRENT_DESKTOP", buf);
+                    free(buf);
                     free(str);
                     free(fname);
                     goto gotit;
-- 
1.9.0

